renderer/src/features/clients/components/__tests__/manage-clients-button.test.tsx

```diff
@@ -0,0 +1,324 @@
+import { describe, it, expect, vi, beforeEach } from 'vitest'
+import { render, screen, waitFor } from '@testing-library/react'
+import { Suspense } from 'react'
+import userEvent from '@testing-library/user-event'
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+import { ManageClientsButton } from '../manage-clients-button'
+import { server } from '@/common/mocks/node'
+import { http, HttpResponse } from 'msw'
+import { mswEndpoint } from '@/common/mocks/msw-endpoint'
+import { PromptProvider } from '@/common/contexts/prompt/provider'
+
+type Recorded = { method: string; url: string; path: string; body?: unknown }
+function startRecording(filter?: (url: string, method: string) => boolean) {
+  const records: Recorded[] = []
+  const onStart = async ({ request }: { request: Request }) => {
+    const method = request.method
+    const url = request.url
+    const path = new URL(url).pathname
+    if (filter && !filter(url, method)) return
+
+    let body: unknown
+    try {
+      if (method !== 'GET' && method !== 'DELETE') {
+        const text = await request.clone().text()
+        body = text
+          ? (() => {
+              try {
+                return JSON.parse(text)
+              } catch {
+                return text
+              }
+            })()
+          : undefined
+      }
+    } catch {
+      // ignore
+    }
+    records.push({ method, url, path, body })
+  }
+  server.events.on('request:start', onStart)
+  return {
+    get: () => records,
+    stop: () => {
+      server.events.removeListener('request:start', onStart)
+    },
+  }
+}
+
```

Comment:
it seems weird to see this implementation here. I like the record payload that we had in Minder, I will add the same feature here in a dedicated util file. I mean it is a middleware

Reply:
yeah, I was thinking also about simply copying the entire msw handling logic from Minder. that could include the automatic mocks even :thinking: 

What do you think about moving that existing system into this repo in a separate PR? Then we can simplify a few of the existing mock-based tests as well :thinking:

Reply:
yep makes sense to do it in a dedicated Pr